server-session/src/main/java/com/iluwatar/sessionserver/App.java

```diff
@@ -75,37 +78,34 @@ public static void main(String[] args) throws IOException {
     server.start();
 
     // Start background task to check for expired sessions
-    sessionExpirationTask();
+    startSessionExpirationTask();
 
     LOGGER.info("Server started. Listening on port 8080...");
   }
 
-  private static void sessionExpirationTask() {
-    new Thread(() -> {
-      while (true) {
-        try {
-          LOGGER.info("Session expiration checker started...");
-          Thread.sleep(SESSION_EXPIRATION_TIME); // Sleep for expiration time
-          Instant currentTime = Instant.now();
-          synchronized (sessions) {
-            synchronized (sessionCreationTimes) {
-              Iterator<Map.Entry<String, Instant>> iterator =
-                  sessionCreationTimes.entrySet().iterator();
-              while (iterator.hasNext()) {
-                Map.Entry<String, Instant> entry = iterator.next();
-                if (entry.getValue().plusMillis(SESSION_EXPIRATION_TIME).isBefore(currentTime)) {
-                  sessions.remove(entry.getKey());
-                  iterator.remove();
-                }
+  private static void startSessionExpirationTask() {
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    Runnable sessionExpirationChecker = () -> {
+      try {
+        LOGGER.info("Session expiration checker started...");
+        Instant currentTime = Instant.now();
+        synchronized (sessions) {
+          synchronized (sessionCreationTimes) {
+            Iterator<Map.Entry<String, Instant>> iterator = sessionCreationTimes.entrySet().iterator();
+            while (iterator.hasNext()) {
+              Map.Entry<String, Instant> entry = iterator.next();
+              if (entry.getValue().plusMillis(SESSION_EXPIRATION_TIME).isBefore(currentTime)) {
+                sessions.remove(entry.getKey());
+                iterator.remove();
               }
             }
           }
-          LOGGER.info("Session expiration checker finished!");
-        } catch (InterruptedException e) {
-          LOGGER.error("An error occurred: ", e);
-          Thread.currentThread().interrupt();
         }
+        LOGGER.info("Session expiration checker finished!");
+      } catch (Exception e) {
+        LOGGER.error("An error occurred: ", e);
       }
-    }).start();
+    };
+    scheduler.scheduleAtFixedRate(sessionExpirationChecker, 0, SESSION_EXPIRATION_TIME, TimeUnit.MILLISECONDS);
```

Comment:
We should also shut down the scheduler when the program exits. Something like this:

```
Runtime.getRuntime().addShutdownHook(new Thread(() -> {
        scheduler.shutdown();
        try {
            if (!scheduler.awaitTermination(60, TimeUnit.SECONDS)) {
                scheduler.shutdownNow();
            }
        } catch (InterruptedException e) {
            scheduler.shutdownNow();
        }
    }));
```